includes/class-wc-stripe-payment-tokens.php

```diff
@@ -321,7 +321,7 @@ public function woocommerce_get_customer_upe_payment_tokens( $tokens, $user_id,
 				if (
 					! isset( $stored_tokens[ $payment_method->id ] ) &&
 					$this->is_valid_payment_method_id( $payment_method->id, $payment_method_type ) &&
-					( $this->is_valid_payment_method_type_for_gateway( $payment_method_type, $gateway_id ) || empty( $gateway_id ) )
+					( empty( $gateway_id ) || $this->is_valid_payment_method_type_for_gateway( $payment_method_type, $gateway_id ) )
```

Comment:
Reversing the empty check order to avoid passing the empty value to `is_valid_payment_method_type_for_gateway`.